synapse/models/math.py

```diff
@@ -0,0 +1,50 @@
+import synapse.lib.module as s_module
+
+class MathModule(s_module.CoreModule):
+
+    def getModelDefs(self):
+        return (('math', {
+            'types': (
+
+                ('math:algorithm', ('guid', {}), {
+                    'doc': 'A mathematical algorithm.'}),
+
+                ('math:algorithm:type:taxonomy', ('taxonomy', {}), {
+                    'interfaces': ('meta:taxonomy',),
+                    'doc': 'A hierarchical taxonomy of algorithm types.'}),
+            ),
+            'edges': (
+
+                (('risk:tool:software', 'uses', 'math:algorithm'), {
+                    'doc': 'The tool uses the algorithm.'}),
+
+                (('it:prod:softver', 'uses', 'math:algorithm'), {
+                    'doc': 'The software uses the algorithm.'}),
+
+                (('file:bytes', 'uses', 'math:algorithm'), {
+                    'doc': 'The file uses the algorithm.'}),
+
+                (('math:algorithm', 'generated', None), {
```

Comment:
This is a new edge...past vs. present tense ("generates")? 

(Most edges are present tense, there are a few "we only know about it b/c it already happened" past tense - e.g., leaked, leveraged.)